services/ens/ensresolver/resolver.go

```diff
@@ -83,20 +84,57 @@ func (e *EnsResolver) OwnerOf(ctx context.Context, chainID uint64, username stri
 		return nil, err
 	}
 
+	nameHash := walletCommon.NameHash(username)
+
 	registry, err := e.contractMaker.NewRegistry(chainID)
 	if err != nil {
 		return nil, err
 	}
 
 	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
-	owner, err := registry.Owner(callOpts, walletCommon.NameHash(username))
+	owner, err := registry.Owner(callOpts, nameHash)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the NameWrapper contract address for the given chain ID
+	nameWrapperAddress, err := walletCommon.GetNameWrapperAddress(chainID)
 	if err != nil {
 		return nil, err
 	}
 
+	// If the owner is the NameWrapper contract, resolve the true owner
+	if owner == nameWrapperAddress {
```

Comment:
super-nit, I think it's more go-idiomatic not to nest logic inside ifs and return early inside it instead. Something like

```suggestion
	if owner != nameWrapperAddress {
		return &owner, nil
	}

	return e.resolveWrappedOwner(ctx, chainID, nameHash, nameWrapperAddress)
```